site/tests/requests/test_record_upgrade_request.py

```diff
@@ -0,0 +1,584 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Zenodo is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Test Zenodo legacy record upgrade request."""
+import pytest
+from invenio_access.permissions import system_identity
+from invenio_rdm_records.proxies import current_rdm_records
+from invenio_rdm_records.records import RDMParent, RDMRecord
+from invenio_records_resources.services.errors import PermissionDeniedError
+from invenio_requests import current_requests_service
+from invenio_requests.proxies import current_events_service
+
+from zenodo_rdm.legacy.requests.utils import submit_record_upgrade_request
+
+
+@pytest.fixture()
+def service(running_app):
+    """The record service."""
+    return current_rdm_records.records_service
+
+
+def _send_post_action(action, request_id, client, headers, expected_status_code):
+    """Send http-post request to perform an action on a request"""
+    response = client.post(
+        f"/requests/{request_id}/actions/{action}",
+        headers=headers,
+        json={},
+    )
+
+    assert response.status_code == expected_status_code
+    return response
+
+
+def _create_legacy_record(identity, record, permission_flags, db, service):
+    """Create and publish a legacy record"""
+    draft = service.create(identity, record)
+    record = service.publish(identity, draft.id)
+
+    # get a UUID of the record
+    record = RDMRecord.pid.resolve(record.id)
```

Comment:
In the tests (not in the code), you can do:
```suggestion
    record = record._record
```